pkg/authorization/api/v1beta1/types.go

```diff
@@ -28,6 +28,9 @@ type PolicyRule struct {
 	Resources []string `json:"resources"`
 	// ResourceNames is an optional white list of names that the rule applies to.  An empty set means that everything is allowed.
 	ResourceNames []string `json:"resourceNames,omitempty"`
+	// NonResourceUrlsSlice is a set of partial urls that a user should have access to.  *s are allowed, but only as the full, final step in the path
```

Comment:
Need a clear description of how this is used in combination with the `Resources`/`ResourceNames` restrictions. Is this required to match requests in addition to Resources/ResourceNames, or instead of?